### ISSUE TITLE
orgFactorsClient.GetFactor always null

### ISSUE THREAD

**Issue by thnk2wn** [okta/oktasdk-csharp#14]
I'm calling orgFactorsClient.GetFactor("okta_sms") and it returns null. In Fiddler I can see this JSON comes back:

> "id": "okta_sms",
    "provider": "OKTA",
    "factorType": "sms",
    "status": "ACTIVE",

but the object is null. The internal implementation looks like this - it has a check on id matching but it doesn't set factor = result or return result so it's always null.

```
public virtual Factor GetFactor(string strMfaType)
    {
      Factor factor = (Factor) null;
      foreach (OktaObject result in (IEnumerable<Factor>) this.GetList((Uri) null, 200, (FilterBuilder) null, SearchType.Filter, (string) null, (string) null, new DateTime?()).Results)
      {
        if (result.Id == strMfaType)
          break;
      }
      return factor;
    }
```

This method is also a bit misleading as it makes me think I'm fetching a single factor but really it's pulling back all of them and then filtering the list down in memory. So it's a bit wasteful from a resource and performance standpoint.

**Comment by thnk2wn** [comment]
For now I'm just working around with:
`smsFactor = orgFactorsClient.GetList().Results.Single(f => f.Id == MfaType.Sms.DisplayName);`

**Comment by raphaellondner-okta** [comment]
Hi Geoff,

Thanks a lot for reporting this bug! I have now fixed the issue and implemented your performance-optimal fix in the latest version available here and in NuGet (v0.3.0.2).

Moving forward, feel free to make any modification you deem beneficial or fix any bug you find by forking this repo, creating a specific branch and submitting a pull request into this repo. We'll make sure to review and respond quickly to any modification you or any other developer submits.

Thanks,

Raphael.
